user_tools/src/spark_rapids_tools/tools/additional_heuristics.py

```diff
@@ -0,0 +1,120 @@
+            RegexPattern.rapids_profile.match,
+            return_directories=True,
+        )
+        if not profile_list:
+            self.logger.warning('No RAPIDS profiles found in output directory: %s', self.output_dir)
+            return pd.DataFrame()
+
+        app_id_paths = find_paths(profile_list[0], RegexPattern.app_id.match, return_directories=True)
+        result_arr = []
+        if not app_id_paths:
+            self.logger.warning('Skipping empty profile: %s', profile_list[0])
+        else:
+            for app_id_path in app_id_paths:
+                try:
+                    app_info_path = os.path.join(app_id_path, self.props.get_value('appInfo', 'fileName'))
+                    app_info = pd.read_csv(app_info_path)
+                    app_name = app_info['appName'].values[0]
+                    app_id = Path(app_id_path).name
+                    # Apply heuristics and determine if the application should be skipped.
+                    # Note: `should_skip` flag can be a combination of multiple heuristic checks.
+                    should_skip = self.heuristics_based_on_spills(app_id_path)
+                    result_arr.append([app_name, app_id, should_skip])
+                except Exception as e:  # pylint: disable=broad-except
+                    self.logger.error('Error occurred while applying additional heuristics. '
+                                      'Reason - %s:%s', type(e).__name__, e)
+        return pd.DataFrame(result_arr, columns=self.props.get_value('resultCols'))
+
+    def heuristics_based_on_spills(self, app_id_path: str) -> bool:
+        """
+        Apply heuristics based on spills to determine if the app can be accelerated on GPU.
+        """
+        # Load stage aggregation metrics (this contains spill information)
+        job_stage_agg_metrics_file = self.props.get_value('spillBased', 'jobStageAggMetrics', 'fileName')
+        job_stage_agg_metrics = pd.read_csv(os.path.join(app_id_path, job_stage_agg_metrics_file))
+        job_stage_agg_metrics = job_stage_agg_metrics[self.props.get_value('spillBased',
+                                                                           'jobStageAggMetrics', 'columns')]
+
+        # Load sql-to-stage information (this contains Exec names)
+        sql_to_stage_info_file = self.props.get_value('spillBased', 'sqlToStageInfo', 'fileName')
+        sql_to_stage_info = pd.read_csv(os.path.join(app_id_path, sql_to_stage_info_file))
+        sql_to_stage_info = sql_to_stage_info[self.props.get_value('spillBased',
+                                                                   'sqlToStageInfo', 'columns')]
+
+        # Identify stages with significant spills
+        spill_threshold_bytes = self.props.get_value('spillBased', 'spillThresholdBytes')
+        stages_with_spills = job_stage_agg_metrics[
+            job_stage_agg_metrics['ID'].str.startswith('stage') &
+            (job_stage_agg_metrics['diskBytesSpilled_sum'] +
```

Comment:
you are adding diskBytesSpilled to memoryBytesSpilled here, that is not correct. on a CPU event log those should not be added.  its a single spill and each of those represent sizes as if they were in memory vs compressed on disk.  
I would likely just stick to one, probably memory unless we have more data since the disk size compression could have more of an effect

Reply:
Thanks @tgravescs. Checking the memory spill makes sense. I will update the logic.